File: app/src/main/java/org/codepond/commitbrowser/commitdetail/CommitDetailActivity.java
package org.codepond.commitbrowser.commitdetail;

import android.os.Bundle;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.LinearLayoutManager;
import android.view.MenuItem;

import org.codepond.commitbrowser.R;
import org.codepond.commitbrowser.common.recyclerview.ItemAdapter;
import org.codepond.commitbrowser.common.ui.BaseActivity;
import org.codepond.commitbrowser.databinding.CommitDetailActivityBinding;

public class CommitDetailActivity extends BaseActivity<CommitDetailViewModel, CommitDetailActivityBinding> {
    public static final String EXTRA_COMMIT_SHA = "extra_commit_sha";

    private String sha;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        sha = getIntent().getStringExtra(EXTRA_COMMIT_SHA);
        setTitle(sha);

        LinearLayoutManager layoutManager = new LinearLayoutManager(this);
        getBinding().recyclerview.setLayoutManager(layoutManager);
        getBinding().recyclerview.setItemAnimator(new DefaultItemAnimator());
        getBinding().recyclerview.addItemDecoration(new DividerItemDecoration(this, DividerItemDecoration.VERTICAL));
        getBinding().recyclerview.setAdapter(new ItemAdapter(getViewModel().getItems(), null));
    }

    @Override
    protected void onStart() {
        super.onStart();
        if (getViewModel().getItems().size() == 0) {
            subscribe(getViewModel().loadDetail(sha));
        }
    }

    @Override
    protected Class<CommitDetailViewModel> getViewModelClass() {
        return CommitDetailViewModel.class;
    }

    @Override
    protected int getLayoutId() {
        return R.layout.commit_detail_activity;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            onBackPressed();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/org/codepond/commitbrowser/commitdetail/CommitDetailActivity.kt
package org.codepond.commitbrowser.commitdetail

import android.os.Bundle
import android.view.MenuItem
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import org.codepond.commitbrowser.R
import org.codepond.commitbrowser.common.ui.BaseActivity
import org.codepond.commitbrowser.databinding.CommitDetailActivityBinding

class CommitDetailActivity : BaseActivity<CommitDetailViewModel, CommitDetailActivityBinding>() {
    override val viewModelClass: Class<CommitDetailViewModel> = CommitDetailViewModel::class.java
    override val layoutId: Int = R.layout.commit_detail_activity

    private lateinit var sha: String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        sha = intent.getStringExtra(EXTRA_COMMIT_SHA)
        title = sha

        binding.recyclerview.layoutManager = LinearLayoutManager(this)
        binding.recyclerview.itemAnimator = DefaultItemAnimator()
        binding.recyclerview.addItemDecoration(
            DividerItemDecoration(
                this,
                DividerItemDecoration.VERTICAL
            )
        )
        viewModel.commitDetail.observe(this, Observer {
            // Update recyclerview adapter
        })
        viewModel.loadDetail(sha)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            onBackPressed()
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    companion object {
        const val EXTRA_COMMIT_SHA = "extra_commit_sha"
    }
}
